src/OrbitBase/include/OrbitBase/Overloaded.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_OVERLOADED_H_
+#define ORBIT_BASE_OVERLOADED_H_
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+#include <utility>
+
+namespace orbit_base_internal {
+
+// The class wraps a function pointer, providing `operator()`
+template <typename FunctionPtr>
+class FunctionPtrWithParen {
+ public:
+  FunctionPtrWithParen(FunctionPtr t) : function_ptr_(t) {}  // NOLINT(google-explicit-constructor)
+
+  template <typename In>
+  auto operator()(In&& in) const -> decltype(std::invoke(std::declval<FunctionPtr>(), in)) {
```

Comment:
The return type also needs the std::forward because this can influence if it compiles are not. (If FunctionPtr only takes r-values for example)

Reply:
Yeah, missed that. Thanks.